test/src/main/java/org/apache/accumulo/test/functional/HalfClosedTabletIT.java

```diff
@@ -0,0 +1,279 @@
+  // DEBUG: MinC failed sleeping 11891 ms before retrying
+  // DEBUG: MinC failed sleeping 43156 ms before retrying
+  // DEBUG: MinC failed sleeping 179779 ms before retrying
+  @Test
+  public void testBadIteratorOnStack() throws Exception {
+
+    // In this scenario the table is using an iterator for minc that is throwing an exception.
+    // This test ensures that the Tablet can still be unloaded normally by taking if offline
+    // after the bad iterator has been removed from the minc configuration.
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String tableName = getUniqueNames(1)[0];
+      final var tops = c.tableOperations();
+
+      tops.create(tableName);
+      final var tid = TableId.of(tops.tableIdMap().get(tableName));
+
+      IteratorSetting is = new IteratorSetting(30, BadIterator.class);
+      c.tableOperations().attachIterator(tableName, is, EnumSet.of(IteratorScope.minc));
+
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation(new Text("r1"));
+        m.put("acf", tableName, "1");
+        bw.addMutation(m);
+      }
+
+      c.tableOperations().flush(tableName, null, null, false);
+
+      UtilWaitThread.sleepUninterruptibly(30, TimeUnit.SECONDS);
+
+      // minc should fail, so there should be no files
+      FunctionalTestUtils.checkRFiles(c, tableName, 1, 1, 0, 0);
+
+      // tell the server to take the table offline
+      tops.offline(tableName);
+
+      try {
+        // We are expecting this to fail, the table should not be taken offline
+        // because the bad iterator will prevent the minc from completing.
+        Wait.waitFor(() -> countHostedTablets(c, tid) == 0L, 120_000);
+        fail("Zero hosted tablets for table.");
+      } catch (IllegalStateException e) {}
+
+      // minc should fail, so there should be no files
+      FunctionalTestUtils.checkRFiles(c, tableName, 1, 1, 0, 0);
+
+      // remove the bad iterator. The failing minc task is in a backoff retry loop
+      // and should pick up this change on the next try
+      c.tableOperations().removeIterator(tableName, BadIterator.class.getSimpleName(),
+          EnumSet.of(IteratorScope.minc));
+
+      // Minc should have completed successfully
+      Wait.waitFor(() -> tabletHasExpectedRFiles(c, tableName, 1, 1, 1, 1), 340_000);
+
+      // Taking the table offline should succeed normally
+      tops.offline(tableName);
+      Wait.waitFor(() -> countHostedTablets(c, tid) == 0L, 340_000);
+    }
+  }
+
+  public static void setInvalidClassLoaderContextPropertyWithoutValidation(ServerContext context,
+      TableId tableId) {
+    TablePropKey key = TablePropKey.of(context, tableId);
+    context.getPropStore().putAll(key,
+        Map.of(Property.TABLE_CLASSLOADER_CONTEXT.getKey(), "invalid"));
+  }
+
+  public static void removeInvalidClassLoaderContextPropertyWithoutValidation(ServerContext context,
```

Comment:
This comment is just a general question. Wondering if we need this method, it validation performed when a prop is deleted that prevents removal of a bad context config?  If so it seems like that could be a general problem.

Reply:
I could try running it without this method. I think you might be correct, that it's not needed. It could also be removed and tested in a follow-on.